CMakeLists.txt

```diff
@@ -190,6 +190,11 @@ target_compile_features(sleigh_settings INTERFACE
   cxx_std_17
 )
 
+target_compile_definitions(sleigh_settings INTERFACE
+  "SLEIGH_SPEC_INSTALL_DIR=\"${CMAKE_INSTALL_PREFIX}/share/sleigh/\""
+  "SLEIGH_SPEC_BUILD_DIR=\"${CMAKE_CURRENT_BINARY_DIR}/Ghidra\""
+)
+
```

Comment:
This is a good start if the user has built and installed sleigh into the same paths, but I'd really like to support relocatable installations, like where you can build everything, tar it up, extract the tar to an arbitrary prefix and still be able to relatively easily find the spec files without modifications.

This would require the `sleigh-lift` executable to know where it is located (maybe [`cpplocate`](https://github.com/cginternals/cpplocate) or equivalent would help). If someone is building a new binary, then we should expose the location of the spec files through a CMake variable in the installed `sleighConfig.cmake` file.

The above doesn't need to be implemented in this PR, but it's something to keep in mind.

Reply:
@tetsuo-cpp can you use a configure command instead of doing things as remill did with these? The macro way works (and is how it's done in remill, and I regret this). I'll share an example of configuration on Slack.

Reply:
2d632ec

I think this does the trick.

Reply:
Can you configure a private header inside of your `support` directory, so that the main `Support.cpp` is part of the codebase and subject to IDE syntax coloring and stuff.

Similarly, can you make the search paths `std::filesystem::path`? Then `gSearchPaths` (renamed to `gDefaultSearchPaths`) can be exported in `Support.h` as:

```c++
extern std::vector<std::filesystem::path> gDefaultSearchPaths;

std::optional<std::filesystem::path>
FindSpecFile(std::string_view file_name,
             const std::vector<std::filesystem::path> &search_paths=gDefaultSearchPaths);
```

Then in Support.cpp:
```c++
const std::vector<std::filesystem::path> gDefaultSearchPaths{
    // Derived from the installation
    kSleighSpecInstallDir,
    // Derived from the build
    kSleightSpecBuildDir,
    // Common install locations
    "/usr/local/share/sleigh",
    "/usr/share/sleigh",
    "/share/sleigh"
};

std::optional<std::filesystem::path>
FindSpecFile(std::string_view file_name,
             const std::vector<std::filesystem::path> &search_paths) {
  // Try paths supplied by the caller first
  for (const auto &path : search_paths) {
    auto file_path = FindSpecFileInSearchPath(file_name, path);
    if (file_path) {
      return file_path;
    }
  }
  // Cannot find the spec file
  return {};
}
```

So support wouldn't search any default paths as a fallback because they would be in `search_paths` as a default-initialized parameter.

Reply:
That doesn't seem very ergonomic to me. If you want to add to more paths to search, then you need to do something like this:
```c++
auto search_paths = sleigh::gDefaultSearchPaths;
search_paths.push_back(path1);
search_paths.push_back(path2);
const auto path = sleigh::FindSpecFile(file_name, search_paths);
```

Reply:
My thinking was that if you knew where it was, then you wouldn't use the default paths. The other way of looking at it is that it seems surprising that it might go and search all over the place, even when you specify a particular list. Worst case, they don't need to copy and extend it; they can instead to do two calls, e.g.

```c++
auto maybe_path = sleigh::FindSpecFile(file_name, my_search_paths);
if (!maybe_path) {
  maybe_path = sleigh::FindSpecFile(file_name);
}
```

Reply:
Makes sense, you're right.

952881a

Reply:
Regarding this:

> Can you configure a private header inside of your support directory, so that the main Support.cpp is part of the codebase and subject to IDE syntax coloring and stuff.

What IDE are you using? I just tried with CLion and it seems to highlight `Support.cpp` correctly. And jumping to definition from the `sleigh-lift` example takes me to the right code.

Reply:
Eclipse CDT, but where I don't let it touch compile_commands.json, and I use out-of-tree builds. Generally, if there are auto-generated files, I prefer them to be `#include`d so that I can introduce a new search path to my preferences, and have all the actual source all in one place.